bubble_sort_by.rb

```diff
@@ -0,0 +1,18 @@
+def bubble_sort_by(array)
+  return array if array.length <= 1
+
+  sum = array.length - 1
+  loop do
+    break if sum <= 0
+
+    sum -= 1
+    array.each do |n|
+      x = array.index(n)
+      next unless array[x + 1]
+
+      array[x], array[x + 1] = array[x + 1], array[x] if yield(array[x], array[x + 1]).positive?
+    end
+  end
+  array
+end
+bubble_sort_by(%w[hi hello hey hhee holapol]) { |left, right| left.length - right.length }
```

Comment:
In line 18 you have this code `bubble_sort_by(%w[hi hello hey hhee holapol]) { |left, right| left.length - right.length }` this code is debugging code. You use it to test if the code works but that's debugging code. 
Please delete it or if you want, keep it in another branch.